admin/modules/config/badwords.php

```diff
@@ -18,85 +18,119 @@
 		}
-	}
-	else
-	{
-		if(!isset($parser) || !is_object($parser))
+
+		if(!$errors)
 		{
-			require_once MYBB_ROOT."inc/class_parser.php";
-			$parser = new postParser;
+			$new_badword = array(
+				"badword" => $db->escape_string($mybb->input['badword']),
+				"regex" => $mybb->get_input('regex', MyBB::INPUT_INT),
+				"replacement" => $db->escape_string($mybb->input['replacement'])
+			);
+
+			$bid = $db->insert_query("badwords", $new_badword);
+
+			$plugins->run_hooks("admin_config_badwords_add_commit");
+
+			// Log admin action
+			log_admin_action($bid, $mybb->input['badword']);
+
+			$cache->update_badwords();
+			flash_message($lang->success_added_bad_word, 'success');
+			admin_redirect("index.php?module=config-badwords");
 		}
-	
-		$badword = $parser->generate_regex($badword);
 	}
 
-	// Don't allow certain badword replacements to be added if it would cause an infinite recursive loop.
-	if(@preg_match('#'.$badword.'#is', $mybb->input['replacement']))
-	{
-		$errors[] = $lang->error_replacement_word_invalid;
-	}
+	$page->add_breadcrumb_item($lang->add_bad_word);
+	$page->output_header($lang->bad_words." - ".$lang->add_bad_word);
 
-	if(!$errors)
-	{
-		$new_badword = array(
-			"badword" => $db->escape_string($mybb->input['badword']),
-			"regex" => $mybb->get_input('regex', MyBB::INPUT_INT),
-			"replacement" => $db->escape_string($mybb->input['replacement'])
-		);
+	$sub_tabs['badwords'] = array(
+		'title' => $lang->bad_word_filters,
+		'description' => $lang->bad_word_filters_desc,
+		'link' => "index.php?module=config-badwords"
+	);
 
-		$bid = $db->insert_query("badwords", $new_badword);
+	$sub_tabs['add_badword'] = array(
+		'title' => $lang->add_bad_word,
+		'description' => $lang->add_bad_word_desc,
+		'link' => "index.php?module=config-badwords&amp;action=add"
+	);
 
-		$plugins->run_hooks("admin_config_badwords_add_commit");
+	$page->output_nav_tabs($sub_tabs, "add_badword");
 
-		// Log admin action
-		log_admin_action($bid, $mybb->input['badword']);
+	$form = new Form("index.php?module=config-badwords&amp;action=add", "post", "add");
 
-		$cache->update_badwords();
-		flash_message($lang->success_added_bad_word, 'success');
-		admin_redirect("index.php?module=config-badwords");
-	}
-	else
+	if($errors)
 	{
-		$mybb->input['action'] = '';
+		$page->output_inline_error($errors);
 	}
+
+	$form_container = new FormContainer($lang->add_bad_word);
+	$form_container->output_row($lang->bad_word." <em>*</em>", $lang->bad_word_desc, $form->generate_text_box('badword', $mybb->get_input('badword'), array('id' => 'badword')), 'badword');
+	$form_container->output_row($lang->replacement, $lang->replacement_desc, $form->generate_text_box('replacement', $mybb->get_input('replacement'), array('id' => 'replacement')), 'replacement');
+	$form_container->output_row($lang->regex, $lang->regex_desc, $form->generate_yes_no_radio('regex', $mybb->get_input('regex'), array('id' => 'regex')), 'regex');
```

Comment:
It should use `$mybb->get_input('regex', MyBB::INPUT_INT)`, the purpose being that enabling the regex setting on by default could cause confusion when the filter isn't an actual regex (i.e. newies or people who don't read).